packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -864,6 +879,23 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
     return new Import(scope, id);
   }
 
+  /**
+   * Imports an existing autoscaling group.
+   */
+  public static fromAutoScalingGroupAttributes(scope: Construct, id: string, attrs: AutoScalingGroupAttributes): IAutoScalingGroup {
+    class Import extends AutoScalingGroupBase {
+      public autoScalingGroupName = attrs.autoScalingGroupName
+      public autoScalingGroupArn = Stack.of(this).formatArn({
+        service: 'autoscaling',
+        resource: 'autoScalingGroup:*:autoScalingGroupName',
+        resourceName: attrs.autoScalingGroupName,
+      });
+      public readonly osType = ec2.OperatingSystemType.UNKNOWN;
+      public readonly grantPrincipal = attrs.grantPrincipal;
```

Comment:
`grantPrincipal` here should also be renamed to `role`, as in the above comment.